### PR TITLE
Makes the do_* family check QDELETED on stuff

### PR DESCRIPTION
**Pull request by Cyberboss**
Oversights

### CODE REVIEW

**Comment by XDTM** [comment]
i guess you could remove the !X checks, since they're included in the QDELETED macro?

File: code/__HELPERS/mobs.dm [hunk 1]
```diff
@@ -266,11 +266,11 @@ Proc for attack log creation, because really why not
 			drifting = 0
 			Uloc = user.loc
 
-		if(!user || user.stat || user.weakened || user.stunned  || (!drifting && user.loc != Uloc) || (extra_checks && !extra_checks.Invoke()))
+		if(QDELETED(user) || user.stat || user.weakened || user.stunned  || (!drifting && user.loc != Uloc) || (extra_checks && !extra_checks.Invoke()))
 			. = 0
 			break
 
-		if(Tloc && (!target || Tloc != target.loc))
+		if(Tloc && (QDELETED(target) || Tloc != target.loc))
```
Comment (RemieRichards) [inline code/__HELPERS/mobs.dm]: what if someone deleted ```Tloc```?

**Comment by RemieRichards** [comment]
@Cyberboss sorry, that was more of a hint for the entire proc, not just that one variable :S

**Comment by Cyberboss** [comment]
ahh!

**Comment by RemieRichards** [comment]
@Cyberboss I found your dark secret... MR CYOND.

**Comment by Cyberboss** [comment]
SSHHHHHH!

**Comment by RemieRichards** [comment]
You should also stop talking to yourself on that, it's ~weird~

**Comment by Cyberboss** [comment]
Futureproof notes

**Comment by RemieRichards** [comment]
No, you're actually having conversations! stahp!

**Pull request merged**